redundant-bmc/src/manager.cpp

```diff
@@ -89,9 +107,38 @@ sdbusplus::async::task<> Manager::startup()
         updateRole(determineRole());
     }
 
+    co_await postStartupClearFOInProgress();
+
     spawnRoleHandler();
+}
 
-    co_return;
+// NOLINTNEXTLINE
+sdbusplus::async::task<> Manager::postStartupClearFOInProgress()
+{
+    if (!redundancyInterface.failover_in_progress())
+    {
+        co_return;
+    }
+
+    // If true, this property is used by both BMCs to determine their roles.
+    // This BMC will have already used it.  Check the other BMC already has
+    // as well and then it can be set to false and removed from the persistent
+    // data.
+
+    lg2::info(
+        "Waiting for sibling to get role and then clearing failover in progress");
+
+    co_await providers->getSibling().waitForSiblingRole();
+
+    redundancyInterface.failover_in_progress(false);
+    try
+    {
+        data::remove(data::key::failoverInProgress);
```

Comment:
I’m wondering if there could be issues when syncing RBMC persisted data to another BMC. Since we don’t sync during FO, if one or both BMCs reboot, we would still rely on the unsynced persisted FO-Inprogress value for role claiming. I just want to confirm if this flow, or do you think we should separate this data into a file that isn’t synced?

Reply:
This app's persistent data file data.json isn't synced, it's just the local value.